code/__HELPERS/unique_ids.dm

```diff
@@ -0,0 +1,44 @@
+// Unique Datum Identifiers
+
+// Basically, a replacement for plain \refs that ensure the reference still
+// points to the exact same datum/client, but doesn't prevent GC like tags do.
+
+// An unintended side effect of the way UIDs are formatted is that the locate()
+// proc will ignore the number and attempt to locate the reference. I consider
+// this a feature, since it means they're conveniently backwards compatible.
+
+// Turns this:
+//   var/myref = "\ref[mydatum]"
+//   var/datum/D = locate(myref)
+// into this:
+//   var/myUID = mydatum.UID()
+//   var/datum/D = locateUID(myUID)
+
+var/global/next_unique_datum_id = 1
+
+// /client/var/tmp/unique_datum_id = null
+
+/datum/proc/UID()

```

Comment:
Please use the existing `<a href='?src=[REF(src)]` instead of adding these files